src/Storages/StorageMergeTree.cpp

```diff
@@ -1066,8 +1065,13 @@ Pipe StorageMergeTree::alterPartition(
         switch (command.type)
         {
             case PartitionCommand::DROP_PARTITION:
-                checkPartitionCanBeDropped(command.partition);
-                dropPartition(command.partition, command.detach, query_context);
+                if (command.part)
+                {
+                    /// TODO(nv) what would be a good check here?
```

Comment:
Any suggestion for a sensible check here @alexey-milovidov ?

Reply:
https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/MergeTree/MergeTreeData.h#L875?